qemu.mk

```diff
@@ -157,12 +217,38 @@ update_rootfs: update_rootfs-common
 ################################################################################
 # Run targets
 ################################################################################
+
+# OP-TEE expects to know how many core will be booted
+ifdef CFG_TEE_CORE_NB_CORE
+QEMU_SMP := $(CFG_TEE_CORE_NB_CORE)
+else
+QEMU_SMP ?= 1
+CFG_TEE_CORE_NB_CORE := $(QEMU_SMP)
+export CFG_TEE_CORE_NB_CORE
+endif
+
 .PHONY: run
 # This target enforces updating root fs etc
 run: all
 	$(MAKE) run-only
 
 .PHONY: run-only
+
+ifeq ($(USE_ATF),y)
+run-only: get-binaries
+get-binaries:
+	ln -sf $(ARM_TF_OUT)/bl1.bin $(BINARIES_PATH)/bl1.bin
+	ln -sf $(ARM_TF_OUT)/bl2.bin $(BINARIES_PATH)/bl2.bin
+	ln -sf $(ROOT)/out/bios-qemu/bios.bin $(BINARIES_PATH)/bl33.bin
+	ln -sf $(OPTEE_OS_HEADER_V2_BIN) $(BINARIES_PATH)/bl32.bin
+	ln -sf $(OPTEE_OS_PAGER_V2_BIN) $(BINARIES_PATH)/bl32_extra1.bin
+	ln -sf $(OPTEE_OS_PAGEABLE_V2_BIN) $(BINARIES_PATH)/bl32_extra2.bin
+
+BIOS_PATH := $(BINARIES_PATH)/bl1.bin
+else
+BIOS_PATH := $(ROOT)/out/bios-qemu/bios.bin
```

Comment:
Do we need to keep the old bios hack?

Reply:
Hmm, no. Actually, this p-r is almost a rfc. This is why I kept both. But in the end, we should use only 1 scheme. I'll update the p-r, discarding the old boot scheme support.